Doxyfile

```diff
@@ -10,7 +10,8 @@ EXTRACT_ALL            = YES
 EXTRACT_PRIVATE        = YES
 EXTRACT_LOCAL_CLASSES  = NO
 GENERATE_LATEX         = NO
-ENABLE_PREPROCESSING   = NO
+ENABLE_PREPROCESSING   = YES
```

Comment:
The trigger for making this change was a build failure, actually. 
Without it documentation for `OUTPUT_DECODE_RESULTS_FOR_UT` 
https://github.com/mbronk/IRremoteESP8266/blob/4707a4d0a236f6c71953626f87d2998640c4d4ad/src/IRac.cpp#L70 yields
```
/IRac.cpp:93: error: found documented #define but ignoring it because ENABLE_PREPROCESSING is NO.
 (warning treated as error, aborting now)
```

---
Now, this is a poor reason to change it, of course... though I figured it is a good change on its own (hence detached to separate comment), as it makes Doxygen `#ifdef` aware (which is also why I had to change in a few other files, as it begun to complain of functions being wrongly documented (rightfully so).
I have these changes decoupled to a separate commit (https://github.com/mbronk/IRremoteESP8266/commit/03767d800c019aff15cc26e2a39e3282c4fd5525) so may detach to a separate PR or skip completely (though on the latter, I'd argue it might be a good direction to enable it eventually).
Let me know which one you prefer.

Reply:
Honestly, I'd love it to handle preprocessor stuff better/correctly, but it doesn't for most of the library.
Having the documentation favour one processor vs. the  other e.g. ESP8266  vs ESP32 isn't really good IMHO.
I'd rather have this bit of code skipped doxygen-wise than break some of the other documentation assumptions.

If you have a better solution, I'm all ears.

Reply:
> Having the documentation favour one processor vs. the other e.g. ESP8266 vs ESP32 isn't really good IMHO.

Agreed. On the flip side, I guess it's already happening somewhat anyway. Ex.:
https://crankyoldgit.github.io/IRremoteESP8266/doxygen/html/classIRrecv.html#a8fe4d26ef1f863db1db9994fed5fc209
documents https://github.com/crankyoldgit/IRremoteESP8266/blob/55ce00262bcace187c97be14aca6536fa53bbd99/src/IRrecv.cpp#L267-L269

...which is ESP32-specific (though the documentation is deliberately written this way).

Since it's a bigger topic, in order to move this one forward, I'll **revert the Doxy changes for now**, and possibly reintroduce later as its own PR if I can think of a way of making it more proper.

BTW. the `ESP32=1` is not really doing anything useful, it's the `ENABLE_PREPROCESSING=YES` which makes it more "_natural_" (less ```///@cond``` etc.), but I agree this likely requires more caution to introduce (merely flipping it to `YES` makes a few errors appear due to how certain functions are documented (commit 03767d800c019aff15cc26e2a39e3282c4fd5525 shows these).

Reply:
As discussed. Let move that discussion/change to a different PR. No need to hold this up.